### DISCUSSION TITLE
Use sounds from other skins in current skin in use

### DISCUSSION THREAD

**Discussion by gustarmartins** [ppy/osu#18314] [Ideas]
I would like to use the hitsounds from the classic skin, with my skin. Would be good if there were an easy way to do this on Lazer. 

Unfortunatelly I dislike a lot the hitsounds on the new default modern skin, so an option to ignore skin sounds, like on stable, wouldn't resolve this, and I dislike the hitsounds of my skin too (almost all skins nowadays have these silly tictac hitsounds).

 So basically it'd be cool if I could use the default classic skin hitsounds with my skin on Lazer without having to move files, etc.

**Comment by Theighlin** [comment]
Afaik it's planned that users will be able to access all loaded skins' assets from the built-in skin editor. You will then be able to replace them and move them to any skin you want.

**Comment by MaxKruse** [comment]
To bump this, also potentially give another insight into this:

On stable, it is possible to NOT USE (edit: typo) SKIN SAMPLES, while still being able to enable map hitsounds. As far as i can see, this this not possible in lazer.

As hitsounds for maps usually are designed around stable's (what is called "classic" in lazer's skin select) base-samples, which get enhanced by the mapper's custom samples if they so choose, the experience of playing lazer vs stable is still not the same for me.

Is something like "use Skin XYZ's samples" a planned option, or perhaps a toggle to disable skin samples like there is in stable? 
I would find it extremely tedious to manually edit skin's samples everytime i try a new skin but dont want to ruin my audio-experience.

**Reply by peppy** [reply]
Can you link a map where this sounds especially bad for reference?

But yes, your issue would be fixed by something like this discussion is looking to change.

**Reply by MaxKruse** [reply]
https://osu.ppy.sh/beatmapsets/1720983#osu/3517113

small video showing the difference (dont mind the performance, windowed for recording):  

https://user-images.githubusercontent.com/10248041/208234392-5e84d339-f8ed-4186-be2e-0dbc736ffa83.mp4

As well as some skins in question  
[_Shield 2.zip](https://github.com/ppy/osu/files/10251356/_Shield.2.zip)
[KamuiHD.zip](https://github.com/ppy/osu/files/10251357/KamuiHD.zip)
[-+ Seoul v9 Pineapple Color](https://shigeskln.s-ul.eu/aZMjYmod)

**Comment by MaxKruse** [comment]
I was looking into implementing this, and have some UI/UX Questions, and also some assumptions:

Assumption 1: This should be a global override. I would assume almost no player interested in this feature would want to change it on a per-skin basis, making the skin-editor for this use-case an unneeded barrier

Assumption 2: Players coming from stable want to experience equal behaviour to stable. This means to enable the (selected) skin hitsounds first, and falling back to 2013. This is especially important because not every map provides a full sampleset of every possible sample, some for example only provide certrain drum additions, while keeping the rest to the "default" (which is the 2013 classic skin's samples)

This made me consider 2 new skin options in the main setting, see below:

![image](https://user-images.githubusercontent.com/10248041/227712658-a1c2d3ad-c048-411d-aa0f-a181b9d8139b.png)

 At least to me, this does seem cluttered and slightly confusing for a player, but for sake of usability i couldnt come up with a better solution. Any feedback or ideas are welcome, i will hold back onto PRing until this is figured out.

**Reply by ItsShamed** [reply]
I think that the best way to display this, is to make the skin editor match the behaviour of the beatmap editor and have its own workflow. 
By that I mean that you would be able to change the metadata like in the beatmap editor, and you would have sections to customise certain parts of your skin (e.g. hitsounds, combo colours, gameplay cursor like proposed in #22938). Putting everything in the settings is definitely not the way to do it, and you witnessed it.
Of course, everything I said requires some initial design, and implementation which takes a lot of time (especially for the initial design when you have to wait for flyte to respond lol), which is probably why nobody attempted to do this before.

**Reply by peppy** [reply]
Agree. Skin editor needs a sample skinning mode, where you can copy samples from one skin to another, and also define different samples for each ruleset.

It's obviously not the easy or quick path, but I think it's best to wait until we can do it this way rather than adding it to the main settings is proposed. While it's probably amicable to add it to settings, it will likely snowball into more people thinking it's okay to add similar settings like "use cursor from x skin" until we end up with the biggest clusterfuck of a UX ever.
